pkgs/tools/networking/dropbear/default.nix

```
@@ -12,24 +27,33 @@ stdenv.mkDerivation rec {
     sha256 = "0fy5ma4cfc2pk25mcccc67b2mf1rnb2c06ilb7ddnxbpnc85s8s8";
   };
 
-  dontDisableStatic = enableStatic;
+  patches = [
+    # Allow sessions to inherit the PATH from the parent dropbear.
+    # Otherwise they only get the usual /bin:/usr/bin kind of PATH
+    ./pass-path.patch
+  ];
 
+  dontDisableStatic = enableStatic;
   configureFlags = lib.optional enableStatic "LDFLAGS=-static";
 
-  CFLAGS = "-DSFTPSERVER_PATH=\\\"${sftpPath}\\\"";
+  buildInputs = [ zlib ] ++ lib.optionals enableStatic [ glibc.static zlib.static ];
+
+  CFLAGS = lib.pipe (lib.attrNames dflags) [
+    (builtins.map (name: "-D${name}=\\\"${dflags.${name}}\\\""))
+    (lib.concatStringsSep " ")
+  ];
 
   # https://www.gnu.org/software/make/manual/html_node/Libraries_002fSearch.html
   preConfigure = ''
-    makeFlags=VPATH=`cat $NIX_CC/nix-support/orig-libc`/lib
+    makeFlagsArray=(
+      VPATH=`cat $NIX_CC/nix-support/orig-libc`/lib
+      PROGRAMS="${lib.concatStringsSep " " [ "dropbear" "dbclient" "dropbearkey" "dropbearconvert" ] ++ lib.optionals enableSCP ["scp"]}"
```

```suggestion
      PROGRAMS="${lib.concatStringsSep " " ([ "dropbear" "dbclient" "dropbearkey" "dropbearconvert" ] ++ lib.optionals enableSCP ["scp"])}"
```

I think we are missing those.